<?xml version="1.0"?>

<launch>
   <group ns="robOTTO/vision">
      <node pkg="nodelet" type="nodelet" name="vision_nodelet_manager"  args="manager" output="screen"/>

      <node pkg="nodelet" type="nodelet" name="FirstFilter" args="load vision/FilterOnMat vision_nodelet_manager" output="screen">
         <param name="input_topic" value="/camera/image/rgb_raw"/>
         <param name="output_topic" value="filtered_image"/>
      </node>

      <node pkg="nodelet" type="nodelet" name="SecondFilter" args="load vision/FilterOnMat vision_nodelet_manager" output="screen">
         <param name="input_topic" value="filtered_image"/>
         <param name="output_topic" value="worked_image"/>
      </node>

      <node pkg="nodelet" type="nodelet" name="LastFilter" args="load vision/FilterOnMat vision_nodelet_manager" output="screen">
         <param name="input_topic" value="worked_image"/>
         <param name="output_topic" value="output_image"/>
      </node>
   </group>
</launch>
